File: USACO/Contest/Bronze/2021January/herd.py
alphabet=input()
word=input()
(countword,res,countalpha)=(0,1,0)
while countword<len(word):
    if alphabet[countalpha]==word[countword]:
        countword+=1
    countalpha+=1
    if countalpha==26:
        countalpha=0
        res+=1
print(res)
    

File: USACO/Contest/Bronze/2021January/herd.cpp
#include <fstream>
#include <iostream>
#include <vector>
#include <string>
using namespace std;
int main(){
    string alph,word;
    cin>>alph>>word;
    int res=1;
    for (int i=1;i!=word.size();i++){
        auto a=alph.find(word[i-1]);
        auto b=alph.find(word[i]);
        if(a>=b){
            res++;
        }

    }
    cout<<res<<"\n";
    
}